flake.nix

```diff
@@ -1,7 +1,7 @@
 {
   description = "K Kore Language Haskell Backend";
   inputs = {
-    nixpkgs.url = "nixpkgs/nixos-23.05";
+    nixpkgs.url = "github:nixos/nixpkgs";
```

Comment:
this may not be the best idea as its quite easy to accidentally bump nixpkgs on any update to the lock file, which has caused issues in the past.... we should definitely set up the common dependency before merging this (if everything works on this version of nixpkgs)

Reply:
Noted; will revert my change! Thanks